File: PYTHON/Program/Car.py
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 4 
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

from Vehicle import Vehicle
# Car adalah child dari Vehicle
class Car(Vehicle): # atribut private untuk class car:
    __jumlah_kursi = 0
    __jumlah_pintu = 0
    
    # konstruktor
    def __init__(self, jumlah_kursi = 0, jumlah_pintu = 0, plat_nomor = "", merk = "", tahun_produksi = 0, warna = ""):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        self.__jumlah_kursi = jumlah_kursi
        self.__jumlah_pintu = jumlah_pintu
    
    #  getter dan setter
    def set_jumlah_kursi(self, jumlah_kursi): self.__jumlah_kursi = jumlah_kursi
    def get_jumlah_kursi(self): return self.__jumlah_kursi
    
    def set_jumlah_pintu(self, jumlah_pintu): self.__jumlah_pintu = jumlah_pintu
    def get_jumlah_pintu(self): return self.__jumlah_pintu
    

File: CPP/Program/Car.cpp
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 4
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

#pragma once
#include <bits/stdc++.h>
using namespace std;
#include "Vehicle.cpp"

class Car : public Vehicle
{
    // private attributes
    private :
        int jumlah_kursi;
        int jumlah_pintu;
    // constructor
    public :
        Car()
        {

        }  
        // Konstruktor
        Car(int jumlah_kursi, int jumlah_pintu, string plat_nomor, string merk, int tahun_produksi, string warna) : Vehicle(plat_nomor, merk, tahun_produksi, warna) 
        {
            this->jumlah_kursi = jumlah_kursi;
            this->jumlah_pintu = jumlah_pintu;
        }

        // Getter dan setter
        void set_jumlah_kursi(int jumlah_kursi) { this->jumlah_kursi = jumlah_kursi; }
        int get_jumlah_kursi() { return jumlah_kursi; }

        void set_jumlah_pintu(int jumlah_pintu) { this->jumlah_pintu = jumlah_pintu; }
        int get_jumlah_pintu() { return jumlah_pintu; }

        //  destructor
        ~Car(){}
};
